vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -66,34 +63,97 @@ def _apply_filters(
     return data_frame
 
 
-def _apply_filter_interaction(
-    data_frame: pd.DataFrame,
-    ctds_filter_interaction: List[CallbackTriggerDict],
-    target: str,
+def _apply_chart_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
 ) -> pd.DataFrame:
-    for ctd in ctds_filter_interaction:
-        if not ctd["value"]:
+    ctd_click_data = ctd_filter_interaction["clickData"]
+    if not ctd_click_data["value"]:
+        return data_frame
+
+    source_chart_id: ModelID = ctd_click_data["id"]
+    source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    try:
+        custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
+    except KeyError as exc:
+        raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
+
+    customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+    for action in source_chart_actions:
+        if target not in action.function["targets"]:
             continue
+        for custom_data_idx, column in enumerate(custom_data_columns):
+            data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
 
-        source_chart_id = ctd["id"]
-        source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    return data_frame
 
-        try:
-            custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
-        except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
 
-        customdata = ctd["value"]["points"][0]["customdata"]
+def _get_parent_vizro_table(_underlying_table_id: str) -> Table:
+    from vizro.models import Table
 
-        for action in source_chart_actions:
-            if target not in action.function["targets"]:
-                continue
+    for _, table in model_manager._items_with_type(Table):
+        if table._underlying_table_id == _underlying_table_id:
+            return table
+    raise KeyError(f"No parent Vizro.Table component found for underlying table with id {_underlying_table_id}.")
+
+
+def _apply_table_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+) -> pd.DataFrame:
+    ctd_active_cell = ctd_filter_interaction["active_cell"]
+    ctd_derived_viewport_data = ctd_filter_interaction["derived_viewport_data"]
+    if not ctd_active_cell["value"] or not ctd_derived_viewport_data["value"]:
+        return data_frame
```

Comment:
I see now what data is filtered, and it's not just the state. Understood. So then every targets dataframe that share the same column id with the source is filtered:
- by the target column
- for the value that is clicked

Is this sufficient? Have users requested to customize this?

Reply:
I agreed with J about this behaviour because we don't need to handle artificially added `"custom_data"` dash.datatable parameter (as we do in vizx). Once we enable targeting filter interaction with the following format (you proposed):
`"source_column.target_id.target_column"` (or something similar), we will adjust calculations in this util function.